railties/lib/rails/commands/credentials/credentials_command.rb

```diff
@@ -38,8 +38,13 @@ def show
 
       private
         def ensure_master_key_has_been_added
-          master_key_generator.add_master_key_file
-          master_key_generator.ignore_master_key_file
+          if !Rails.application.credentials.content_path.exist?
+            master_key_generator.add_master_key_file
+            master_key_generator.ignore_master_key_file
+          else
+            say missing_credentials_message
```

Comment:
So we're saying credentials are missing if the content path exists? That doesn't make sense to me.

Why won't my earlier suggestion do?

```ruby
def ensure_master_key_has_been_added
  if Rails.application.credentials.key.nil? && !Rails.application.credentials.content_path.exist?
    master_key_generator.add_master_key_file
    master_key_generator.ignore_master_key_file
  end
end
```

Reply:
@kaspth 
There are the 2 differences from yours

1. `Rails.application.credentials.key.nil?` is omitted  
because this is already checked when `ensure_master_key_has_been_added` is called.

2. exits and show `missing_credentials_message`  if it steps into the else branch

Without exit, if you run `rails credentials:edit`,  you edit `/tmp/2293.credentials.yml` like the following screen shot.

<img width="895" alt="2018-08-12 12 26 39" src="https://user-images.githubusercontent.com/7948153/43998206-27277966-9e2b-11e8-9caa-cb41b1b59db9.png">

 If nothing changes in it, the message `New credentials encrypted and saved.` shows but `credentials.yml.enc` is not saved, which I check the time stamp. But adding some changes and save it,  the error below occurs:
```
$ EDITOR=vi bundle exec rails credentials:edit
Traceback (most recent call last):
	19: from bin/rails:4:in `<main>'
	18: from bin/rails:4:in `require'
	17: from /vagrant/rails/railties/lib/rails/commands.rb:18:in `<top (required)>'
	16: from /vagrant/rails/railties/lib/rails/command.rb:46:in `invoke'
	15: from /vagrant/rails/railties/lib/rails/command/base.rb:65:in `perform'
	14: from /var/lib/gems/2.5.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	13: from /var/lib/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	12: from /var/lib/gems/2.5.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	11: from /vagrant/rails/railties/lib/rails/commands/credentials/credentials_command.rb:26:in `edit'
	10: from /vagrant/rails/railties/lib/rails/command/helpers/editor.rb:26:in `catch_editing_exceptions'
	 9: from /vagrant/rails/railties/lib/rails/commands/credentials/credentials_command.rb:27:in `block in edit'
	 8: from /vagrant/rails/railties/lib/rails/commands/credentials/credentials_command.rb:55:in `change_credentials_in_system_editor'
	 7: from /vagrant/rails/activesupport/lib/active_support/encrypted_file.rb:54:in `change'
	 6: from /vagrant/rails/activesupport/lib/active_support/encrypted_file.rb:68:in `writing'
	 5: from /vagrant/rails/activesupport/lib/active_support/encrypted_configuration.rb:29:in `write'
	 4: from /vagrant/rails/activesupport/lib/active_support/encrypted_file.rb:49:in `write'
	 3: from /vagrant/rails/activesupport/lib/active_support/encrypted_file.rb:75:in `encrypt'
	 2: from /vagrant/rails/activesupport/lib/active_support/message_encryptor.rb:151:in `encrypt_and_sign'
	 1: from /vagrant/rails/activesupport/lib/active_support/message_encryptor.rb:169:in `_encrypt'
/vagrant/rails/activesupport/lib/active_support/message_encryptor.rb:169:in `key=': key must be 16 bytes (ArgumentError)
```

That's thy I implement like this. I think `exit` is needed and then some messages are necessary.
But if you don't like to use `missing_credentials_message` to show the message, I will modify it. I would like a suggestion what the message is.

Reply:
Doesn't make sense to me that there's a conditional both within and outside ensure_master_key_has_been_added. They should both go in there.

Sure, but we've already got the line `say Rails.application.credentials.read.presence || missing_credentials_message`, won't this be in a state where the credentials can't be read?